docs/flow/reference/components/prometheus.remote_write.md

```diff
@@ -0,0 +1,349 @@
+provided, it must be set to one of the following strings:
+
+* `"TLS10"` (TLS 1.0)
+* `"TLS11"` (TLS 1.1)
+* `"TLS12"` (TLS 1.2)
+* `"TLS13"` (TLS 1.3)
+
+### queue_config block
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`capacity` | `number` | Number of samples to buffer per shard. | `2500` | no
+`min_shards` | `number` | Minimum amount of concurrent shards sending samples to the endpoint. | `1` | no
+`max_shards` | `number` | Maximum number of concurrent shards sending samples to the endpoint. | `200` | no
+`max_samples_per_send` | `number` | Maximum number of samples per send. | `500` | no
+`batch_send_deadline` | `duration` | Maximum time samples will wait in the buffer before sending. | `"5s"` | no
+`min_backoff` | `duration` | Initial retry delay. The backoff time gets doubled for each retry. | `"30ms"` | no
+`max_backoff` | `duration` | Maximum retry delay. | `"5s"` | no
+`retry_on_http_429` | `bool` | Retry when an HTTP 429 status code is received. | `false` | no
+
+Each queue then manages a number of concurrent _shards_ which is responsible
+for sending a fraction of data to their respective endpoints. The number of
+shards is automatically raised if samples are not being sent to the endpoint
+quickly enough. The range of permitted shards can be configured with the
+`min_shards` and `max_shards` arguments.
+
+Each shard has a buffer of samples it will keep in memory, controlled with the
+`capacity` argument. New metrics won't be read from the WAL unless there is at
+least one shard that is not at maximum capcity.
+
+The buffer of a shard will be flushed and sent to the endpoint either after the
+shard reaches the number of samples specified by `max_samples_per_send` or the
+duration specified by `batch_send_deadline` has elapsed since the last flush
+for that shard.
+
+Shards will infinitely retry requests which fail with an `HTTP 5xx` status code
+until the requests succeeds. The delay between retries can be customized with
+the `min_backoff` and `max_backoff` arguments.
+
+The `retry_on_http_429` argument can optionally configure shards to retry
+requests which fail with an `HTTP 429` status code; other requests which fail
+due to a `HTTP 4xx` status code are never retried. When `retry_on_http_429` is
+enabled, `Retry-After` response headers from the servers are honored.
+
+### metadata_config block
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`send` | `bool` | Controls wheter metric metadata is sent to the endpoint. | `true` | no
+`send_interval` | `duration` | How frequently metric metadata is sent to the endpoint. | `"1m"` | no
+`max_samples_per_send` | `number` | Maximum number of metadata samples to send to the endpoint at once. | `500` | no
+
+### wal block
+
+The `wal` block customizes the WAL used to locally store metrics before they
+are sent to the configured set of endpoints.
```

Comment:
Kudos for the detailed explanations 🎉 

Do we want half a sentence to explicitly call out that the WAL is used as a buffer in case there's intermittent network issues and populating the cache after a restart? Not sure if it's worth spelling out or if it's widely understood.